src/commands/createNewProject.ts

```diff
@@ -204,7 +204,7 @@ async function createJavaFunctionProject(outputChannel: OutputChannel, functionA
     return appName;
 }
 
-export async function createNewProject(outputChannel: OutputChannel, functionAppPath?: string, openFolder: boolean = true, ui: IUserInterface = new VSCodeUI()): Promise<void> {
+export async function createNewProject(outputChannel: OutputChannel, telemetryReporter: TelemetryReporter | undefined, functionAppPath?: string, openFolder: boolean = true, ui: IUserInterface = new VSCodeUI()): Promise<void> {
```

Comment:
Why not just make this optional?

Reply:
I refactored this anyways, but there's a difference between 'optional' parameters and 'required parameters that could be undefined'. The reporter should _always_ be passed (it just happens to be undefined if there was an error creating it).